File: frontend/apps/pm-app/lib/features/auth/data/models/property_manager.dart
class PropertyManager {
  final String email;
  final String? phoneNumber;
  final String businessName;
  final String businessAddress;
  final String city;
  final String state;
  final String zipCode;
  final String accountStatus;
  final String setupProgress;

  PropertyManager({
    required this.email,
    this.phoneNumber,
    required this.businessName,
    required this.businessAddress,
    required this.city,
    required this.state,
    required this.zipCode,
    required this.accountStatus,
    required this.setupProgress,
  });

  factory PropertyManager.fromJson(Map<String, dynamic> json) {
    return PropertyManager(
      email: json['email'] as String? ?? '',
      phoneNumber: json['phone_number'] as String?,
      businessName: json['business_name'] as String? ?? '',
      businessAddress: json['business_address'] as String? ?? '',
      city: json['city'] as String? ?? '',
      state: json['state'] as String? ?? '',
      zipCode: json['zip_code'] as String? ?? '',
      accountStatus: json['account_status'] as String? ?? '',
      setupProgress: json['setup_progress'] as String? ?? '',
    );
  }
}

File: backend/shared/go-models/property_manager.go
package models

import (
	"time"

	"github.com/google/uuid"
)

type PMAccountStatusType string

const (
	PMAccountStatusIncomplete PMAccountStatusType = "INCOMPLETE"
	PMAccountStatusActive     PMAccountStatusType = "ACTIVE"
)

// PM-specific setup progress aliases for clarity in PM contexts
const (
	PMSetupProgressAwaitingInfo SetupProgressType = "AWAITING_INFO"
	PMSetupProgressDone                 SetupProgressType = "DONE"
)

type PropertyManager struct {
	Versioned

	ID              uuid.UUID           `json:"id"`
	Email           string              `json:"email"`
	PhoneNumber     *string             `json:"phone_number,omitempty"`
	TOTPSecret      string              `json:"totp_secret,omitempty"`
	BusinessName    string              `json:"business_name"`
	BusinessAddress string              `json:"business_address"`
	City            string              `json:"city"`
	State           string              `json:"state"`
	ZipCode         string              `json:"zip_code"`
	AccountStatus   PMAccountStatusType `json:"account_status"`
	SetupProgress   SetupProgressType   `json:"setup_progress"`
	CreatedAt       time.Time           `json:"created_at"`
	UpdatedAt       time.Time           `json:"updated_at"`
	DeletedAt       *time.Time          `json:"deleted_at,omitempty"`
}

// ----- concurrency helpers -----
func (pm *PropertyManager) GetID() string { return pm.ID.String() }
